src/app/services/technical-record/technical-record.service.ts

```diff
@@ -172,6 +170,23 @@ export class TechnicalRecordService {
     return this.http.put<VehicleTechRecordModel>(url, body, { responseType: 'json' });
   }
 
+  calculateTechRecordToUpdate(newVehicleTechRecord: VehicleTechRecordModel, recordToArchiveStatus?: StatusCodes): TechRecordModel {
+    let newTechRecord;
+
+    if (recordToArchiveStatus === StatusCodes.PROVISIONAL) {
+      newTechRecord = newVehicleTechRecord.techRecord.find(techRecord => techRecord.statusCode === StatusCodes.PROVISIONAL);
+    } else {
+      newTechRecord = newVehicleTechRecord.techRecord.find(techRecord => techRecord.statusCode === StatusCodes.CURRENT);
+      if (!newTechRecord) {
+        newTechRecord = newVehicleTechRecord.techRecord.find(techRecord => techRecord.statusCode === StatusCodes.PROVISIONAL);
+      }
+    }
+
+    if (!newTechRecord) throw new Error(`Cannot find a provisional or current to update`);
+
+    return newTechRecord;
+  }
+
```

Comment:
```suggestion
  calculateTechRecordToUpdate(newVehicleTechRecord: VehicleTechRecordModel, recordToArchiveStatus?: StatusCodes): TechRecordModel {
    const filteredRecords = newVehicleTechRecord.techRecord.filter(techRecord => techRecord.statusCode !== StatusCodes.ARCHIVED);
    const newTechRecord =
      filteredRecords.find(techRecord => techRecord.statusCode === (recordToArchiveStatus ?? StatusCodes.CURRENT)) ??
      filteredRecords.find(techRecord => techRecord.statusCode === StatusCodes.PROVISIONAL);

    if (!newTechRecord) throw new Error(`Cannot find a provisional or current to update`);

    return newTechRecord;
  }
  
```
Don't really know if it's nicer, but passes all the tests

Reply:
Like this, but going to go with Taha's suggestion below instead, thank you!